azblob/url_block_blob.go

```diff
@@ -108,6 +115,23 @@ func (bb BlockBlobURL) GetBlockList(ctx context.Context, listType BlockListType,
 	return bb.bbClient.GetBlockList(ctx, listType, nil, nil, ac.pointers(), nil)
 }
 
+// CopyFromURL synchronously copies the data at the source URL to a block blob, with sizes up to 256 MB.
+// For more information, see https://docs.microsoft.com/en-us/rest/api/storageservices/copy-blob-from-url.
+func (bb BlockBlobURL) CopyFromURL(ctx context.Context, source url.URL, metadata Metadata,
+	srcac ModifiedAccessConditions, dstac BlobAccessConditions, srcContentMD5 []byte) (*BlobCopyFromURLResponse, error) {
+
+	srcIfModifiedSince, srcIfUnmodifiedSince, srcIfMatchETag, srcIfNoneMatchETag := srcac.pointers()
+	dstIfModifiedSince, dstIfUnmodifiedSince, dstIfMatchETag, dstIfNoneMatchETag := dstac.ModifiedAccessConditions.pointers()
+	dstLeaseID := dstac.LeaseAccessConditions.pointers()
+
+	return bb.blobClient.CopyFromURL(ctx, source.String(), nil, metadata, AccessTierNone,
+		srcIfModifiedSince, srcIfUnmodifiedSince,
+		srcIfMatchETag, srcIfNoneMatchETag,
+		dstIfModifiedSince, dstIfUnmodifiedSince,
+		dstIfMatchETag, dstIfNoneMatchETag,
+		dstLeaseID, nil, srcContentMD5)
+}
```

Comment:
On the one hand, this seems questionable to have on a block blob URL, since according to the docs, it's under the general blob operations.

On the other hand, it can only target block blobs.

So this is mostly just a question of "how well can the user locate this?".

@JohnRusk thoughts?

Reply:
The location of the REST doc is probably wrong. It should be under Block Blob operations instead.